b2share/modules/remotes/views.py

```diff
@@ -0,0 +1,193 @@
+        if not destination_file_url:
+            raise UserError("missing required destination_file_url parameter")
+
+        b2drop_urlbase = url_for('remotes.b2drop', path='', _external=True)
+        b2drop_urlbase_relative = urlparse(b2drop_urlbase).path
+        if source_remote_url.startswith(b2drop_urlbase):
+            b2drop_path = source_remote_url[len(b2drop_urlbase):]
+        elif source_remote_url.startswith(b2drop_urlbase_relative):
+            b2drop_path = source_remote_url[len(b2drop_urlbase_relative):]
+        else:
+            raise UserError("bad source_remote_url; currently only supporting b2drop remotes")
+
+        fileBucket_urlbase = url_for('invenio_files_rest.location_api', _external=True)
+        if not destination_file_url.startswith(fileBucket_urlbase):
+            raise UserError("bad destination_file_url; must point to a file bucket object")
+        destination_file_url = destination_file_url[len(fileBucket_urlbase):]
+        destination_file_url = destination_file_url.strip('/')
+        [bucket_id, key] = destination_file_url.split('/', maxsplit=1)
+        if not key or not bucket_id:
+            raise UserError("bad destination_file_url; must be a correct URL "
+                            "pointing to a file in a file bucket object")
+
+        b2drop_client = session.get('b2drop_client')
+        if not b2drop_client:
+            raise UserError("B2DROP remote not initialized")
+
+        stream = b2drop_client.make_stream_object(b2drop_path)
+        return put_file_into_bucket(bucket_id, key, stream, stream.length())
+
+
+def put_file_into_bucket(bucket_id, key, stream, content_length):
+    # TODO: refactor invenio_files_rest to have a proper API and use that one here
+    from invenio_db import db
+    from invenio_files_rest.models import Bucket, ObjectVersion
+    from invenio_files_rest.views import need_bucket_permission
+    from invenio_files_rest.errors import FileSizeError
+
+    bucket = Bucket.get(bucket_id)
+    if bucket is None:
+        abort(404, 'Bucket does not exist.')
+
+    # WARNING: this function should be isomorphic with
+    #          invenio_files_rest.views:ObjectResource.create_object
+    @need_bucket_permission('bucket-update')
+    def create_object(bucket, key):
+        size_limit = bucket.size_limit
+        if size_limit and content_length > size_limit:
+            desc = 'File size limit exceeded.' \
+                if isinstance(size_limit, int) else size_limit.reason
+            raise FileSizeError(description=desc)
+
+        with db.session.begin_nested():
+            obj = ObjectVersion.create(bucket, key)
+            obj.set_contents(
+                stream, size=content_length, size_limit=size_limit)
+        db.session.commit()
+        return obj
+
+    return create_object(bucket, key)
+
+
+blueprint.add_url_rule('/remotes/jobs',
+                       view_func=Jobs.as_view(Jobs.view_name))
```

Comment:
Ok. This might be an issue if someone tries to upload a big file because it would stop the whole server. The scripts in `dockerize`run the server with `b2share run` and not a real webserver such as gunicorn or tornado. Thus it will block the whole testing server.
Can you add a file limit size so that the tests can run smoothly? We will remove it later.